tests/integration/nupic/opf/htmpredictionmodel_serialization_test.py

```diff
@@ -0,0 +1,267 @@
+                          'fields': [(u'c1', 'sum'), (u'c0', 'first')],
+        'hours': 1,
+        'microseconds': 0,
+        'milliseconds': 0,
+        'minutes': 0,
+        'months': 0,
+        'seconds': 0,
+        'weeks': 0,
+        'years': 0},
+    'predictAheadTime': None,
+    'modelParams': {
+      #'inferenceType': 'TemporalAnomaly', # backtracking_tm.py", line 425, in __getattr__: AttributeError: 'TM' object has no attribute 'write'
+      #'inferenceType': 'NontemporalAnomaly', # sdr_classifier.py", line 429, in write - IndexError: deque index out of range
+      #'inferenceType': 'NontemporalClassification', # sdr_classifier.py", line 429, in write: IndexError: deque index out of range
+      'inferenceType': 'NontemporalMultiStep', # sdr_classifier.py", line 429, in write: IndexError: deque index out of range
+        'sensorParams': {
+          'verbosity' : 0,
+            'encoders': {
+              u'consumption':    {  'clipInput': True,
+                                      'fieldname': u'consumption',
+                    'maxval': 100.0,
+                    'minval': 0.0,
+                    'n': 50,
+                    'name': u'c1',
+                    'type': 'ScalarEncoder',
+                    'w': 21},},
+            'sensorAutoReset' : None,
+            },
+        'spEnable': True,
+        'spParams': {
+          'spatialImp' : 'py',
+            'spVerbosity' : 0,
+            'globalInhibition': 1,
+            'columnCount': 512,
+            'inputWidth': 0,
+            'numActiveColumnsPerInhArea': 20,
+            'seed': 1956,
+            'potentialPct': 0.5,
+            'synPermConnected': 0.1,
+            'synPermActiveInc': 0.1,
+            'synPermInactiveDec': 0.005,
+            },
+        'tmEnable' : True,
+        #'tmEnable' : False,
+        'tmParams': {
+          'temporalImp': 'py',
+            'verbosity': 0,
+            'columnCount': 512,
+            'cellsPerColumn': 8,
+            'inputWidth': 512,
+            'seed': 1960,
+            'newSynapseCount': 10,
+            'maxSynapsesPerSegment': 20,
+            'maxSegmentsPerCell': 32,
+            'initialPerm': 0.21,
+            'permanenceInc': 0.1,
+            'permanenceDec' : 0.1,
+            'globalDecay': 0.0,
+            'maxAge': 0,
+            'minThreshold': 4,
+            'activationThreshold': 6,
+            'outputType': 'normal',
+            'pamLength': 1,
+            },
+        'clParams': {
+          'implementation': 'py',
+            'regionName': 'SDRClassifierRegion',
+            'verbosity' : 0,
+            'alpha': 0.005,
+            'steps': '1,5',
+            },
+
+        'anomalyParams': {  u'anomalyCacheRecords': None,
+                            u'autoDetectThreshold': None,
+    u'autoDetectWaitRecords': 2184},
+
+        'trainSPNetOnlyIfRequested': False,
+        },
+}
+
+
+class HTMPredictionModelSerializationTest(unittest.TestCase):
```

Comment:
NOTE: This test is very rudimentary, just enough to demonstrate the failure being addressed by this pull request. The test will evolve in subsequent pull requests related to testing of the serialization of `HTMPredictionModel`.